src/overlays/actors/ovl_En_Ma1/z_en_ma1.c

```diff
@@ -19,45 +36,410 @@ const ActorInit En_Ma1_InitVars = {
+                    ret = 2;
+                    break;
+                default:
+                    ret = 0;
+                    break;
+            }
+            break;
+        case 4:
+        case 5:
+            if (func_80106BC8(globalCtx) != 0) {
+                ret = 2;
+            }
+            break;
+        case 6:
+            if (func_80106BC8(globalCtx) != 0) {
+                ret = 3;
+            }
+            break;
+        case 0:
+        case 1:
+        case 3:
+        case 7:
+        case 8:
+        case 9:
+            ret = 1;
+            break;
+    }
+    return ret;
+}
+
+s32 func_80AA08C4(EnMa1* this, GlobalContext* globalCtx) {
+    if ((this->actor.shape.rot.z == 3) && (gSaveContext.sceneSetupIndex == 5)) {
+        return 1;
+    }
+    if (gSaveContext.linkAge != 1) {
+        return 0;
+    }
+    if (((globalCtx->sceneNum == SCENE_MARKET_NIGHT) || (globalCtx->sceneNum == SCENE_MARKET_DAY)) &&
+        (!(gSaveContext.eventChkInf[1] & 0x10)) && (!(gSaveContext.infTable[8] & 0x800))) {
+        return 1;
+    }
+    if ((globalCtx->sceneNum == SCENE_SPOT15) && (!(gSaveContext.eventChkInf[1] & 0x10))) {
+        if (gSaveContext.infTable[8] & 0x800) {
+            return 1;
+        } else {
+            gSaveContext.infTable[8] |= 0x800;
+            return 0;
+        }
+    }
+    if ((globalCtx->sceneNum == SCENE_SOUKO) && (gSaveContext.nightFlag == 1) && (gSaveContext.eventChkInf[1] & 0x10)) {
+        return 1;
+    }
+    if (globalCtx->sceneNum != SCENE_SPOT20) {
+        return 0;
+    }
+    if ((this->actor.shape.rot.z == 3) && (gSaveContext.nightFlag == 0) && (gSaveContext.eventChkInf[1] & 0x10)) {
+        return 1;
+    }
+    return 0;
+}
+
+void func_80AA0A0C(EnMa1* this) {
+    if (DECR(this->unk_1E2) == 0) {
+        this->unk_1E4 += 1;
+        if (this->unk_1E4 >= 3) {
+            this->unk_1E2 = Math_Rand_S16Offset(0x1E, 0x1E);
+            this->unk_1E4 = 0;
+        }
+    }
+}
+
+void func_80AA0A84(EnMa1* this, UNK_TYPE idx) {
+    f32 frameCount = SkelAnime_GetFrameCount(&D_80AA1678[idx].animation->genericHeader);
+
+    SkelAnime_ChangeAnim(&this->skelAnime, D_80AA1678[idx].animation, 1.0f, 0.0f, frameCount, D_80AA1678[idx].unk_08,
+                         D_80AA1678[idx].transitionRate);
+}
+
+void func_80AA0AF4(EnMa1* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    s16 phi_a3;
+
+    if ((this->unk_1E8.unk_00 == 0) && (this->skelAnime.animCurrentSeg == &D_06008D64)) {
+        phi_a3 = 1;
+    } else {
+        phi_a3 = 0;
+    }
+
+    this->unk_1E8.unk_18 = player->actor.posRot.pos;
+    this->unk_1E8.unk_18.y -= -10.0f;
+
+    func_80034A14(&this->actor, &this->unk_1E8, 0, phi_a3);
+}
+
+void func_80AA0B74(EnMa1* this) {
+    if (this->skelAnime.animCurrentSeg == &D_06008D64) {
+        if (this->unk_1E8.unk_00 == 0) {
+            if (this->unk_1E0 != 0) {
+                this->unk_1E0 = 0;
+                func_800F6584(0, this);
```

Comment:
this function appears to be called by en_ma1, 2 and 3. I would make this &this->actor, and below where it is used again

Reply:
It actually doesn't even take that other arg, which I'll fix in Ma2 after this is merged